src/android/com/microsoftopentechnologies/intellij/wizards/createvm/EndpointStep.java

```diff
@@ -0,0 +1,396 @@
+        }
+
+        @Override
+        public Component getTableCellRendererComponent(JTable jTable, Object value, boolean b, boolean b1, int i, int i1) {
+            setSelectedItem(value);
+
+            return this;
+        }
+    }
+
+    private class ErrorRenderer extends JLabel implements TableCellRenderer  {
+
+        public ErrorRenderer() {
+            setOpaque(false);
+        }
+
+        @Override
+        public Component getTableCellRendererComponent(JTable jTable, Object value, boolean isSelected, boolean hasFocus, int row, int col) {
+            setPreferredSize(new Dimension(getSize().height, 15));
+            setForeground(Color.red);
+            setHorizontalAlignment(CENTER);
+            setFont(getFont().deriveFont(Font.BOLD));
+
+            if(jTable.getModel() != null && jTable.getModel() instanceof EndpointTableModel) {
+                EndpointTableModel endpointTableModel = (EndpointTableModel) jTable.getModel();
+                String errorList = getErrorFromRow(row, endpointTableModel.getData());
+                setToolTipText(errorList);
+                setText(errorList.isEmpty() ? "" : "!");
+            }
+
+            return this;
+        }
+    }
+
+    private class EndpointTableModel extends AbstractTableModel {
+        private Vector<Endpoint> data = new Vector<Endpoint>();
+        private String[] columns = new String[]{ "", "Port name", "Public port", "Private port", "Protocol", "" };
+
+        @Override
+        public int getRowCount() {
+            return data.size();
+        }
+
+        @Override
+        public int getColumnCount() {
+            return columns.length;
+        }
+
+        @Override
+        public String getColumnName(int i) {
+            return columns[i];
+        }
+
+        @Override
+        public boolean isCellEditable(int row, int col) {
+            return (col > 0 && col < 5);
+        }
+        @Override
+        public Object getValueAt(int row, int column){
+            Endpoint endpoint = data.get(row);
+            switch(column) {
+                case 1:
+                    return endpoint.getName();
+                case 2:
+                    return endpoint.getPublicPort();
+                case 3:
+                    return endpoint.getPrivatePort();
+                case 4:
+                    return endpoint.getProtocol();
+                default:
+                    return "";
+            }
+        }
+
+        @Override
+        public void setValueAt(Object object, int row, int col) {
+            Endpoint endpoint = data.get(row);
+            switch(col) {
+                case 1:
+                    endpoint.setName(object.toString());
+                    break;
+                case 2:
+                    try {
+                        int publicPort = Integer.parseInt(object.toString());
+                        endpoint.setPublicPort(publicPort);
+                    } catch (NumberFormatException ex) {}
```

Comment:
Shouldn't we handle the `NumberFormatException` and show an error message to the user? It might make sense to show a non-intrusive error, i.e. an error message on a label on the dialog or something instead of a modal message box.